src/modules/roc_address/io_uri_format.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "roc_address/io_uri.h"
+#include "roc_address/pct.h"
+#include "roc_core/panic.h"
+
+namespace roc {
+namespace address {
+
+namespace {
+
+bool str_append(char* buf, size_t bufsz, size_t& pos, const char* str) {
+    const size_t orig_len = strlen(str);
```

Comment:
I guess that we should ```roc_panic_if(!str)``` to avoid SIGFAULT for strlen(NULL);

Reply:
I did not add it because this an unexported function. I usually prefer to check contracts on module/unit boundaries and avoid them inside module/unit, unless it is somewhat very complex.

On the other hand, why not? Added.